File: benchmarks/pandas/bench_series_log_natural.py
"""Benchmark: Series natural logarithm — np.log / Series.apply(np.log) on 100k-element Series."""
import json
import time
import pandas as pd
import numpy as np

ROWS = 100_000
WARMUP = 5
ITERATIONS = 50

s = pd.Series(np.arange(1, ROWS + 1, dtype=float))

for _ in range(WARMUP):
    np.log(s)

start = time.perf_counter()
for _ in range(ITERATIONS):
    np.log(s)
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "series_log_natural",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))

File: benchmarks/tsb/bench_series_log_natural.ts
/**
 * Benchmark: seriesLog — natural logarithm on a 100k-element Series.
 * Outputs JSON: {"function": "series_log_natural", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */
import { Series, seriesLog } from "../../src/index.ts";

const ROWS = 100_000;
const WARMUP = 5;
const ITERATIONS = 50;

// Positive values to avoid NaN in log
const s = new Series({ data: Array.from({ length: ROWS }, (_, i) => (i % 10000) + 1) });

for (let i = 0; i < WARMUP; i++) {
  seriesLog(s);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  seriesLog(s);
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "series_log_natural",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
